Secret-agency-persistence/src/main/java/cz/fi/muni/pa165/secretagency/dao/DepartmentDaoImpl.java

```diff
@@ -27,7 +27,7 @@ public DepartmentDaoImpl() {
     @Override
     public long getNumberOfEmployees(Long departmentId) {
         Object result = em.createQuery("SELECT COUNT(d) FROM Department d " +
-                "LEFT JOIN d.agents " +
+                "inner JOIN d.agents " +
```

Comment:
Já bych radši ten LEFT JOIN. 😄

Reply:
ale to nefunguje, jelikoz to interne dela OUTER join a ten ikdyz nenajde zadneho employee, tak vraci 1

Reply:
Ani v případě, když se napíše něco jako:

SELECT COUNT(a) FROM Department d " +
                "LEFT JOIN d.agents a" + ...

Reply:
Ale než tohle, tak je asi lepší inner join.

Reply:
`SELECT COUNT(a) FROM Department d LEFT JOIN Agent a ON a.department = d WHERE d.id = :departmentId
`

takto to fici, libi se ti to vic? 😄

Reply:
Ten inner join je hezčí. Pokud je ekvivalentní. SQL jde trochu mimo mě. 😄

Reply:
tak nechavam tak jak to je teda 

jedina zmena je, ze to dela inner misto outer, furt je to left join.

outer join ti prida null hodnoty, tam kde nic nenajde, to znamena, ze kdyz nenajde zadneho agenta, tak to doplni null hodnotama a zustane tam jeden radek